src/formatDate.js

```diff
@@ -51,6 +51,69 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const maxYear = 30;
+
+  const shortYear = 'YY';
+  const longYear = 'YYYY';
+  const twentiethCentury = '20';
+  const ninetiethCentury = '19';
+  const century = 100;
+
+  const oldObjectDate = {};
+  const oldArrayDate = date.split(oldSeparator);
+  const newArrayDate = [];
+  const lengthOfNewDate = 3;
+
+  for (let i = 0; i < oldArrayDate.length; i++) {
+    if (fromFormat[i] === shortYear) {
+      if (+oldArrayDate[i] < maxYear) {
+        oldArrayDate[i] = twentiethCentury + oldArrayDate[i];
+      } else {
+        oldArrayDate[i] = ninetiethCentury + oldArrayDate[i];
+      }
+      fromFormat[i] = longYear;
+    }
+    oldObjectDate[fromFormat[i]] = oldArrayDate[i];
+  }
+
+  for (let i = 0; i < lengthOfNewDate; i++) {
+    if (toFormat[i] === shortYear) {
+      newArrayDate.push(oldObjectDate[longYear] % century);
+    } else {
+      newArrayDate.push(oldObjectDate[toFormat[i]]);
+    }
+  }
```

Comment:
It's not readable at all. Avoid nested "if" statements, also prefer don't use "else" statements, for example, you can write in "if" "continue" statements so it wouldn't proceed after "if"([link to docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Loops_and_iteration#continue_statement)).